service/resource/cloudformationv2/create.go

```diff
@@ -0,0 +1,90 @@
+package cloudformationv2
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/keyv2"
+)
+
+func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
+	cluster, err := keyv2.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	stackInput, err := toCreateStackInput(createChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	_, err = r.Clients.CloudFormation.CreateStack(&stackInput)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	err = r.HostClients.CloudFormation.WaitUntilStackCreateComplete(&cloudformation.DescribeStacksInput{
+		StackName: stackInput.StackName,
+	})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "debug", "creating AWS cloudformation stack: created")
+
+	// Create host post-main stack. It includes the peering routes, which need resources from the
+	// guest stack to be in place before it can be created.
+	err = r.createHostPostStack(cluster)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	customObject, err := keyv2.ToCustomObject(obj)
+	if err != nil {
+		return cloudformation.CreateStackInput{}, microerror.Mask(err)
+	}
+
+	desiredStackState, err := toStackState(desiredState)
```

Comment:
Not a blocker but I wonder if we need to update the StackState struct? So if covers all 3 stacks.

Reply:
Not sure, do we need to update the host stacks? Currently we have there the cross-account access  role (only depends on the guest account id) and the peering routes (depend on the host VPC and the VPCPeeringID). As you suggest maybe better discuss it in a separate issue, will create it.

Reply:
Good point if there is nothing to change then it doesn't need to be in the stack state. I think later we might have to add more to the main host stack but we could do that later.

Yes let's discuss more in the issue.